### PR TITLE
feature: legacy web support

### PR DESCRIPTION
**Pull request by gomball**

### CODE REVIEW

**Comment by gomball** [comment]
resolves #60

File: package.json [hunk 1]
```diff
@@ -15,14 +15,17 @@
   },
   "devDependencies": {
     "@types/jest": "^24.0.15",
+    "browserify": "^16.5.1",
     "jest": "^24.8.0",
+    "mkdirp": "^1.0.4",
```
Comment (Sayan751) [inline package.json:20]:
```suggestion
```
Browserify creates the dir for for you. So you don't need this. See below.

File: package.json [hunk 2]
```diff
@@ -15,14 +15,17 @@
...
     "rimraf": "^2.6.3",
     "ts-jest": "^24.0.2",
+    "tsify": "^4.0.2",
     "typescript": "^3.5.2"
   },
   "scripts": {
     "prebuild": "rimraf dist",
-    "build": "tsc -p tsconfig.cjs.json & tsc -p tsconfig.esm.json & tsc -p tsconfig.dts.json",
+    "build": "tsc -p tsconfig.cjs.json & tsc -p tsconfig.esm.json & tsc -p tsconfig.dts.json & mkdirp dist/browser & browserify -e src/index-browser.ts -p [ tsify --noImplicitAny ] -o dist/browser/index.js",
```
Comment (Sayan751) [inline package.json:28]:
```suggestion
    "build": "tsc -p tsconfig.cjs.json & tsc -p tsconfig.esm.json & tsc -p tsconfig.dts.json & browserify -e src/index-browser.ts -p [ tsify ] -o dist/browser/index.js",
```

As Browserify creates the out dir, the `mkdirp` can be skipped. Also `tsify` automatically reads the `tsconfig.json` file. The tsconfig at root has a whole lot of options, along with the `noImplicitAny` flag set. That can be skipped too.

**Comment by gomball** [comment]
Hi @Sayan751.

On my first approach y used something like `... browserify dist/cjs/index.js > dist/browser/index.js`, so i needed to create the output dir first... mkdirp was left as a residue: my mistake. Thank you for your comments!

All changes are commited

**Comment by Sayan751** [comment]
@techniq LGTM

**Comment by techniq** [comment]
@gomball Thanks for the PR and @Sayan751 thanks for the review.

Published as [6.1.0](https://www.npmjs.com/package/odata-query).  [np](https://github.com/sindresorhus/np) was giving me a failure when bumping the version so I didn't get a github release with changelog on this one.  @gomball Let me know how it works for you.

**Comment by gomball** [comment]
it's strange: I've installer v6.1.0 locally, and the `dist/browser` folder is missing. even more strange is that `package.json`'s build script is updated with the browserify part. Changes are present on new version, but it seems like build failed on browserify command.

Would it be possible that the version bumping failure is due to the merge commit message? I mean: I dont know how np is configured on this project, but the word "Merge" is commonly forbidden by changelog generators..

**Comment by techniq** [comment]
@gomball This was my bad.  It's been a while since I've manually pushed a release (so reliant on CI/CD nowadays).  I had forgotten to `npm install` locally, although I'm surprised the local `npm publish` didn't fail.  I guess the `preversion` hook was failing but it wasn't obvious when I was running `np`.  Anyways, give `6.1.1` a go and let me know.

**Comment by gomball** [comment]
Hi @techniq.

Everything seems fine with 6.1.1: https://js-lrs4nu.stackblitz.io

**Pull request merged**